server/src/main/java/org/opensearch/extensions/ExtensionsOrchestrator.java

```diff
@@ -150,56 +149,52 @@ private void extensionsDiscovery() throws IOException {
+        if (Files.exists(extensionsPath.resolve("extensions.yml"))) {
             try {
                 extensions = readFromExtensionsYml(extensionsPath.resolve("extensions.yml")).getExtensions();
             } catch (Exception e) {
-                throw new IOException("Could not read from extensions.yml");
+                throw new IOException("Could not read from extensions.yml\n " + e);
             }
-        }
-        for (final Path plugin : pluginDirs) {
-            if (Files.isDirectory(plugin)) {
+            for (Extension extension : extensions) {
                 try {
-                    PluginInfo pluginInfo = PluginInfo.readFromProperties(plugin);
-                    String pluginName = pluginInfo.getName();
-                    for (Extension extension : extensions) {
-                        if (extension.getName().equals(pluginName)) {
-                            try {
-                                extensionsSet.add(
-                                    new DiscoveryExtension(
-                                        extension.getName(),
-                                        extension.getUniqueId(),
-                                        // placeholder for ephemeral id, will change with POC discovery
-                                        extension.getUniqueId(),
-                                        extension.getHostName(),
-                                        extension.getHostAddress(),
-                                        new TransportAddress(TransportAddress.META_ADDRESS, Integer.parseInt(extension.getPort())),
-                                        new HashMap<String, String>(),
-                                        Version.fromString(extension.getVersion()),
-                                        pluginInfo
-                                    )
-                                );
-                                logger.debug("Loaded extension: " + extension);
-                                break;
-                            } catch (IllegalArgumentException e) {
-                                logger.error(e.toString());
-                            }
-                        }
-                    }
-
-                } catch (final IOException e) {
-                    throw new IOException("Could not load plugin descriptor " + plugin.getFileName(), e);
+                    extensionsList.add(
+                        new DiscoveryExtension(
+                            extension.getName(),
+                            extension.getUniqueId(),
+                            // placeholder for ephemeral id, will change with POC discovery
+                            extension.getUniqueId(),
+                            extension.getHostName(),
+                            extension.getHostAddress(),
+                            new TransportAddress(InetAddress.getByName(extension.getHostAddress()), Integer.parseInt(extension.getPort())),
```

Comment:
Should we keep it as `TransportAddress.META_ADDRESS`? @saratvemulapalli any thoughts here

Reply:
^ we changed it this morning while troubleshooting the connection errors

Reply:
For additional context

Reply:
Yeah. I'm just trying to think of a better approach here